lms/views/api/refresh.py

```diff
@@ -0,0 +1,38 @@
+from pyramid.view import view_config, view_defaults
+
+from lms.security import Permissions
+
+
+@view_defaults(request_method="POST", permission=Permissions.API, renderer="json")
+class RefreshViews:
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
+
+    @view_config(route_name="canvas_api.oauth.refresh")
+    @view_config(route_name="blackboard_api.oauth.refresh")
+    def get_refreshed_token(self):
+        """
+        Refresh the user's access token.
+
+        Send a request to get a refreshed access token for the authenticated
+        user and save it to the DB.
+        """
+        if self.request.matched_route.name == "canvas_api.oauth.refresh":
+            self.refresh_canvas_access_token(self.request)
+        else:
+            assert self.request.matched_route.name == "blackboard_api.oauth.refresh"
+            self.refresh_blackboard_access_token(self.request)
```

Comment:
It turns out that there's very little in common between what the view does for a Canvas refresh versus a Blackboard refresh. It might be nicer to just have two separate views instead of a single view that switches based on `request.matched_route`